spacelift/internal/structs/user_group.go

```diff
@@ -0,0 +1,12 @@
+package structs
+
+type SpaceAccessRule struct {
+	Space            string `graphql:"space"`
+	SpaceAccessLevel string `graphql:"spaceAccessLevel"`
+}
+
+type UserGroup struct {
```

Comment:
I'm wondering if perhaps a different name here would be more meaningful, for example, `UserGroupBinding`

Reply:
In the UI, we are using the name IdP group mapping.

Reply:
in that case I'm not gonna introduce a third name here, I'll stick with what's used in the GraphQL schema

Reply:
The problem in that case is that the UI and terraform provider would have different names. They are meant to be similar, at least based on naming, because the person implementing the terraform provider will search for resources that match the naming in the UI.
@marcinwyszynski, what do you think from a usability/conformity perspective?

Reply:
@michieldewilde This is a struct representing the GraphQL `type ManagedUserGroup`. If we want consistency, this is the "canonical" name.

Reply:
I'm referring to the name of the resource (commented in the wrong place).
There is a resource with the name `spacelift_user_group`; we are not using the name user group in any other place than the GraphQL API.
Searching in the documentation and our UI, I can't find any reference to a user group.
We are calling it IdP Group Mapping in those places.
This might confuse users.
There will be essentially two different names for the same concept in public-facing systems (three if you count the difference between `UserGroup` and `ManagedUserGroup`).

Reply:
OK, `spacelift_idp_group_mapping` then?

Reply:
Yes.
It would be best to do it this way and add a description to the `ManagedUserGroup` in GraphQL to clarify that this refers to the IdP group mapping in the UI and the terraform provider.